dleq_vrf/src/keys.rs

```diff
@@ -80,7 +80,7 @@ impl<F: PrimeField> SecretScalar<F> {
     }
 
     /// Multiply by a scalar.
-    pub fn mul_by_challenge(&mut self, rhs: &F) -> F {
+    pub fn mul_by_challenge(&self, rhs: &F) -> F {
         let mut lhs = self.clone();
         lhs *= rhs;
```

Comment:
If I move up the `replit` call in `MulAssign<F>` below then we could still mostly merge your PR, which likely makes things less annoying.

I'll think about some trade offs here.  We might want an RNG inside the sk for example.

Reply:
Ain't clear this achieves anything, so nope..